GCEm/__init__.py

```diff
@@ -7,9 +7,201 @@
+
+
+def gp_model(training_params, training_data, data_processors=None,
+             kernel=None, kernel_op='add', active_dims=None, noise_variance=1.,
+             name='', gpu=0):
+    """
+    Create a Gaussian process (GP) based emulator with provided `training_params` (X) and `training_data` (Y).
+
+    The `kernel` is a key parameter in GP emulation and care should be taken in choosing it.
+
+    Parameters
+    ----------
+    training_params: pd.DataFrame
+        The training parameters
+    training_data: iris.cube.Cube or array_like
+        The training data - the leading dimension should represent training samples
+    data_processors: list of GCEm.data_processors.DataProcessor
+        A list of `DataProcessor`s to apply to the data transparently before training. Model output will be
+        un-transformed before being returned from the Emulator.
+    kernel: gpflow.kernels.Kernel or list of str or None
+        The GP kernel to use.  A GPFlow kernel can be specified directly, or a list of kernel names can be provided
+        which will be initialised using the default values (of the correct shape) and combined using `kernel_op`.
+        Alternatively no kernel can be specified and a default will be used.
+    kernel_op : {'add', 'mul'}
+        The operation to perform in order to combine the specified `kernel`s. Only used if `kernel` is a list of strings.
+    noise_variance: float
+        The noise variance to initialise the GP regression model
+    active_dims: list of int or slice or None
+        The dimensions to train the GP over (by default all of the dimensions are used)
+    name: str
+        An optional name for the emulator
+    gpu: int
+        The GPU to use (only applicable for multi-GPU) machines
+
+    Returns
+    -------
+    Emulator
+        A GCEm emulator object which can be trained and sampled from
+
+    """
+
+    from .model_adaptor import GPFlowModel
+    import tensorflow as tf
+    import gpflow
+    from gpflow.config import default_float
+
+    n_params = training_params.shape[1]
+
+    data_processors = data_processors if data_processors is not None else []
+    data_processors.extend([Recast(default_float()), Flatten()])
+    data = CubeWrapper(training_data, data_processors=data_processors)
+
+    if isinstance(kernel, list):
+        kernel = _get_gpflow_kernel(kernel, n_params, active_dims=active_dims, operator=kernel_op)
+    elif kernel is None:
+        # TODO Maybe this should just use the above mechanism?
+        kernel = gpflow.kernels.RBF(lengthscales=[0.5] * n_params, variance=0.01, active_dims=active_dims) + \
+                 gpflow.kernels.Linear(variance=[1.] * n_params, active_dims=active_dims) + \
+                 gpflow.kernels.Polynomial(variance=[1.] * n_params, active_dims=active_dims) + \
+                 gpflow.kernels.Bias(active_dims=active_dims)
+        print("WARNING: Using default kernel - be sure you understand the assumptions this implies")
```

Comment:
I like the warning! Would it make sense to point towards Duvenaud's 'kernel cookbook'? http://www.cs.toronto.edu/~duvenaud/cookbook/ Or at least I think this should be referenced in the docs.

Reply:
Good idea - I reference it in the paper too :-)

Reply:
Nice, I saw that! There's a typo though, his PhD was in 2014 not 2011 :)